api/cedar_metadata_records/views.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import unicode_literals
+import logging
+
+from rest_framework import permissions as drf_permissions
+from rest_framework.exceptions import NotFound
+from rest_framework.generics import ListCreateAPIView, RetrieveAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.renderers import JSONRenderer
+from rest_framework.views import Response
+
+from api.base import permissions as base_permissions
+from api.base.filters import ListFilterMixin
+from api.base.parsers import (
+    JSONAPIMultipleRelationshipsParser,
+    JSONAPIMultipleRelationshipsParserForRegularJSON,
+)
+from api.base.versioning import PrivateVersioning
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_user_auth
+from api.cedar_metadata_records.permissions import CedarMetadataRecordPermission
+from api.cedar_metadata_records.serializers import (
+    CedarMetadataRecordsListSerializer,
+    CedarMetadataRecordsListCreateSerializer,
+    CedarMetadataRecordsDetailSerializer,
+)
+from api.cedar_metadata_records.utils import can_view_record
+from framework.auth.oauth_scopes import CoreScopes
+
+from osf.models import CedarMetadataRecord
+
+logger = logging.getLogger(__name__)
+
+
+class CedarMetadataRecordList(JSONAPIBaseView, ListCreateAPIView, ListFilterMixin):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.CEDAR_METADATA_RECORD_READ]
+    required_write_scopes = [CoreScopes.CEDAR_METADATA_RECORD_WRITE]
+
+    serializer_class = CedarMetadataRecordsListSerializer
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON, )
+    model_class = CedarMetadataRecord
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+    view_category = 'cedar-metadata-records'
+    view_name = 'cedar-metadata-record-list'
+
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return CedarMetadataRecordsListCreateSerializer
+        else:
+            return CedarMetadataRecordsListSerializer
+
+    def get_default_queryset(self):
+        published_records = CedarMetadataRecord.objects.filter(is_published=True)
+        user_auth = get_user_auth(self.request)
+        record_ids = [record.id for record in published_records if can_view_record(user_auth, record)]
```

Comment:
Agreed. I asked a similar question in BE channel and the suggestion was to do a annotation if the logic is not too complicated. Let me take a look at the `AbstractNodeManager.can_view` and see if it is feasible.

Reply:
As discussed in pair, we disable the list view for all records since it is not used by FE at all. More specifically, using `CreateAPIView` instead of `ListCreateAPIView`